Flaky test in CI

Here is an example of a PR with two test runs where the first fails and the second passes: https://app.circleci.com/pipelines/github/argoproj/argo-rollouts?branch=pull%2F524.

The failing test has the following logs: 
```
--- FAIL: TestNewAnalysisRunFromTemplates (0.00s)
    helpers_test.go:483: 
        	Error Trace:	helpers_test.go:483
        	Error:      	Not equal: 
        	            	expected: "my-arg"
        	            	actual  : "my-secret"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-my-arg
        	            	+my-secret
        	Test:       	TestNewAnalysisRunFromTemplates
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x10d1e5a]

goroutine 113 [running]:
testing.tRunner.func1(0xc000278600)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x11cab40, 0x1da8390)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/argoproj/argo-rollouts/utils/analysis.TestNewAnalysisRunFromTemplates(0xc000278600)
	/go/src/github.com/argoproj/argo-rollouts/utils/analysis/helpers_test.go:484 +0x54a
testing.tRunner(0xc000278600, 0x13ac298)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
```

We need to investigate this test and make it consistent.

--- a/utils/analysis/helpers_test.go
+++ b/utils/analysis/helpers_test.go
@@ -460,30 +460,31 @@ func TestNewAnalysisRunFromTemplates(t *testing.T) {
 			},
 		},
 	}}
-	args := []v1alpha1.Argument{
-		{
-			Name:  "my-arg",
-			Value: pointer.StringPtr("my-val"),
-		},
-		{
-			Name: "my-secret",
-			ValueFrom: &v1alpha1.ValueFrom{
-				SecretKeyRef: &v1alpha1.SecretKeyRef{
-					Name: "name",
-					Key:  "key",
-				},
+
+	arg := v1alpha1.Argument{
+		Name:  "my-arg",
+		Value: pointer.StringPtr("my-val"),
+	}
+	secretArg := v1alpha1.Argument{
+		Name: "my-secret",
+		ValueFrom: &v1alpha1.ValueFrom{
+			SecretKeyRef: &v1alpha1.SecretKeyRef{
+				Name: "name",
+				Key:  "key",
 			},
 		},
 	}
+
+	args := []v1alpha1.Argument{arg, secretArg}
 	run, err := NewAnalysisRunFromTemplates(templates, args, "foo-run", "foo-run-generate-", "my-ns")
 	assert.NoError(t, err)
 	assert.Equal(t, "foo-run", run.Name)
 	assert.Equal(t, "foo-run-generate-", run.GenerateName)
 	assert.Equal(t, "my-ns", run.Namespace)
-	assert.Equal(t, "my-arg", run.Spec.Args[0].Name)
-	assert.Equal(t, "my-val", *run.Spec.Args[0].Value)
-	assert.Equal(t, "my-secret", run.Spec.Args[1].Name)
-	assert.NotNil(t, *run.Spec.Args[1].ValueFrom.SecretKeyRef)
+
+	assert.Len(t, run.Spec.Args, 2)
+	assert.Contains(t, run.Spec.Args, arg)
+	assert.Contains(t, run.Spec.Args, secretArg)
 
 	// Fail Merge Args
 	unresolvedArg := v1alpha1.Argument{Name: "unresolved"}

